src/app/components/harvest/components/shared/title.component.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { fakeAsync, tick } from "@angular/core/testing";
+import { FormsModule } from "@angular/forms";
+import { By } from "@angular/platform-browser";
+import { ShallowHarvestsService } from "@baw-api/harvest/harvest.service";
+import { Harvest } from "@models/Harvest";
+import { Project } from "@models/Project";
+import { SpectatorHost, createHostFactory, SpyObject } from "@ngneat/spectator";
+import { generateHarvest } from "@test/fakes/Harvest";
+import { generateProject } from "@test/fakes/Project";
+import { MockProvider } from "ng-mocks";
+import { TitleComponent } from "./title.component";
+
+describe("titleComponent", () => {
+  let spectator: SpectatorHost<TitleComponent>;
+  let mockHarvest: Harvest;
+  let mockProject: Project;
+
+  const createHost = createHostFactory({
+    component: TitleComponent,
+    imports: [
+      FormsModule,
+    ],
+  });
+
+  function setup(): SpyObject<ShallowHarvestsService> {
+    spectator = createHost(
+      `
+      <baw-harvest-title></baw-harvest-title>
+      `,
+      {
+        props: {
+          project: mockProject,
+          harvest: mockHarvest
+        },
+        providers: [
+          MockProvider(ShallowHarvestsService, {}),
+        ],
+      }
+    );
+
+    const mockHarvestApi = spectator.inject<SpyObject<ShallowHarvestsService>>( ShallowHarvestsService as any );
+    mockHarvestApi.updateName = jasmine.createSpy("updateName") as any;
+
+    spectator.detectChanges();
+
+    return mockHarvestApi;
+  }
+
+  const getHarvestTitle = (): HTMLElement =>
+    spectator.query("small");
+
+  const getNameEditButton = (): HTMLElement =>
+    spectator.query("sup");
+
+  const getNameEditInput = (): HTMLInputElement =>
+    spectator.debugElement.query(By.css("input[name='harvestNameInput']")).nativeElement;
+
+  beforeEach(() => {
+    mockHarvest = new Harvest(generateHarvest({ status: "metadataReview" }));
+    mockProject = new Project(generateProject());
+  });
+
+  it("should create", () => {
+    setup();
+    expect(spectator.component).toBeInstanceOf(TitleComponent);
+  });
+
+  it("should display the name of the project and harvest", () => {
+    setup();
+    const expectedTitle = `Project: ${mockProject.name} | Harvest: ${mockHarvest.name}`;
+
+    expect(getHarvestTitle().innerText).toEqual(expectedTitle);
```

Comment:
does this test pass? if it does i'd say there's something wrong.... look at that string template